src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java

```diff
@@ -141,6 +142,11 @@ public void importEntries(List<BibEntry> entriesToImport, boolean shouldDownload
         // Remember the selection in the dialog
         preferences.getFilePreferences().setDownloadLinkedFiles(shouldDownloadFiles);
 
+        Optional<Path> targetDirectory = databaseContext.getFirstExistingFileDir(preferences.getFilePreferences());
+        if (targetDirectory.isEmpty()) {
+            dialogService.showErrorDialogAndWait(Localization.lang("Import entries"), Localization.lang("File directory needs to be set or does not exist.\nPlease save your library and check the preferences."));
```

Comment:
Yes, I made it general because saving a library is anyway a good practice that should be imbibed in users. Secondly, I thought why only specifically warn in case of pdfs? There too the warning could be completely removed, the entries anyway get imported, and the pdf is accessible by clicking the pdf icon (it would just not be saved to a directory). So I thought either don't warn at all (like other fetchers) or warn whenever importing entries to maintain uniformity.

Reply:
It is only about the first save ever! Not subsequent saves.

Reply:
The PDF is accessible by the URL, not by the file system.

Reply:
> The PDF is accessible by the URL, not by the file system.

Hmm...I see now. In that case there should be two different warning texts. Or I'll look later if I can restrict this only to PDFs.

Reply:
Okay there is a simple way to do that - declaring a ```static boolean dialogShown``` and using it inside the ```execute()``` method of ```DownloadLinkedFileAction.java```. Good thing - now only PDFs will trigger this. The tradeoff - now it will only be shown once for each session, and not if next time another set of entries are imported.
Which fix would you prefer?

Reply:
I think the latter is better - one warning is enough for a user during a session so it's not a big tradeoff, and no warning in case of no PDF.

Reply:
> One could make it even more user friendly if there was no warning text at all - and all fetchers just put the link in case there is no file directory.
> 

Oh so it'd be better if we removed the warning itself? That too is doable. Let me see.